alacritty/src/config/window.rs

```diff
@@ -74,10 +74,18 @@ impl Default for WindowConfig {
 impl WindowConfig {
     #[inline]
     pub fn dimensions(&self) -> Option<Dimensions> {
-        if self.dimensions.columns.0 != 0
-            && self.dimensions.lines != 0
-            && self.startup_mode != StartupMode::Maximized
-        {
+        let is_columns_set = self.dimensions.columns.0 != 0;
+        let is_lines_set = self.dimensions.lines != 0;
+
+        // Throw warning if only one of columns, lines is set
+        if (is_lines_set) ^ (is_columns_set) {
+            warn!(
+                target: LOG_TARGET_CONFIG,
+                "Both lines and columns must be set for window.dimensions to take effect"
+            );
```

Comment:
Should this really be the message when startup mode is set to maximized? Wouldn't it be more accurate to mention that they are both ignored in that case?

Reply:
Yeah, this warning should only be emitted if the options are actually applied.